### PR TITLE
[Keyboard] Add Typface Univers

### PR DESCRIPTION
**Pull request by itsme-zeix**
## Description

Added support for Typface Univers.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zeix/typface/univers/matrix_diagram.md
Comment (dunk2k) [inline keyboards/zeix/typface/univers/matrix_diagram.md:1]: 1. F4 and 2u Backspace keys are labelled with the same matrix co-ordinate, could this be corrected?
2. Could right hand co-ordinate be changed from A-N to 0-9,A-E? to be in uniform with other `matrix_diagram.md` files

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/zeix/typface/univers/rules.mk [hunk 1]
```diff
@@ -0,0 +1 @@
+# This file intentionally left blank
```
Comment (dunk2k) [inline keyboards/zeix/typface/univers/rules.mk:1]:
```suggestion
# This file intentionally left blank

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/zeix/typface/univers/matrix_diagram.md
Comment (itsme-zeix) [reply keyboards/zeix/typface/univers/matrix_diagram.md:1]: Yup, I will correct it. Thanks for the review!

**Review by drashna — APPROVED** [review]

**Comment by dunk2k** [comment]
To confirm to how build markers have been reworked as of most recent breaking changes, can:
1. `rules.mk` be removed
2. `info.json` re-named to `keyboard.json`

File: keyboards/zeix/typface/univers/info.json [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                {"matrix": [3, 4], "x": 4.75, "y": 3.5},
+                {"matrix": [3, 5], "x": 5.75, "y": 3.5},
+                {"matrix": [3, 6], "x": 6.75, "y": 3.5},
+                {"matrix": [3, 7], "x": 7.75, "y": 3.5},
+                {"matrix": [3, 8], "x": 8.75, "y": 3.5},
+                {"matrix": [3, 9], "x": 9.75, "y": 3.5},
+                {"matrix": [3, 10], "x": 10.75, "y": 3.5},
+                {"matrix": [3, 11], "x": 11.75, "y": 3.5},
+                {"matrix": [3, 13], "x": 12.75, "y": 3.5, "w": 2.25},
+                {"matrix": [3, 12], "x": 15.25, "y": 3.5},
```
Comment (dunk2k) [inline keyboards/zeix/typface/univers/info.json:88]: correct key placement for ISO Hash, this does mean ANSI Enter key width is shortened to accomdate
```suggestion
                {"matrix": [3, 12], "x": 12.75, "y": 3.5},
                {"matrix": [3, 13], "x": 13.75, "y": 3.5, "w": 1.25},
                
```

File: keyboards/zeix/typface/univers/info.json [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        "rows": ["GP20", "GP10", "GP9", "GP8", "GP1", "GP0"]
+    },
+    "processor": "RP2040",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x0001",
+        "vid": "0x4C28"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/zeix/typface/univers/info.json:30]:
```suggestion
        "LAYOUT_all": {
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
Based on [store page](https://typface.com/products/univers), this PCB comes in to two variants - can the directory structure of this PR be change to reflect variants available.

**Comment by itsme-zeix** [comment]
> To confirm to how build markers have been reworked as of most recent breaking changes, can:
> 
> 1. `rules.mk` be removed
> 2. `info.json` re-named to `keyboard.json`

Will make these changes, thank you for notifying me about the breaking changes!

> Based on [store page](https://typface.com/products/univers), this PCB comes in to two variants - can the directory structure of this PR be change to reflect variants available.

The firmware supports both the solder and hotswap variant, though the hotswap variant does not include ISO compatibility - in that case, is there still a need to duplicate the files?

File: keyboards/zeix/typface/univers/univers.json
Comment (dunk2k) [inline keyboards/zeix/typface/univers/univers.json:1]: can this file be renamed, literally, to `keyboard.json`?

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
> The firmware supports both the solder and hotswap variant, though the hotswap variant does not include ISO compatibility - in that case, is there still a need to duplicate the files?

Thankfully, duplication can be avoided by:
1. Placing settings and configuration used by both variants in a directory above the proposed `solder` and `hotswap` (sub-)directories. This would include an `info.json` file that has layouts both variants support (i.e. `LAYOUT_ansi`, `LAYOUT_ansi_split_bs`).
2. Solder directory's `keyboard.json` would contain the compatible ISO `"layouts"` and `"community layouts", and the current `LAYOUT_all` layout.

This proposed directory restructure would mean a keymap (sub-directory) and differing PID for each variant.

**Pull request closed without merging**